docs/guides/workplace_bot_doc.mdx

```diff
@@ -0,0 +1,213 @@
+    
+3.  Open up [](http://localhost:4040/)[http://localhost:4040/](http://localhost:4040/) in your browser. Here is where you can see requests sent and received by your webhook.
+    
+4.  Create a new Workplace group chat with your favorite coworkers and your bot, and trigger your bot by calling one of your macros such as “#m lgtm”
+    
+5.  Use [](http://localhost:4040/)[http://localhost:4040/](http://localhost:4040/) and console to debug as needed
+    
+## Limitations
+One major quirk I ran into while building this was the bot must be added to the group chat to work, which means:
+
+1.  If you forget to add the bot to your chat before you try to use the #m command it won’t work
+    
+2.  You cannot use the bot in a direct message. In order for you to use the bot you would have to create a new group chat with yourself, the person you want to message and the bot.
+    
+3.  The bot can not be used in any chat that is associated with a group. Workplace has the ability to add a chat to a workplace group. This comes with nice features such as automatic membership and keeps both your posts and chats in one place. When you try to directly add the bot to the group/chat, the bot does not appear in the list of options when you hit Add people. If you unlink the chat and the group using admin options, you can then add the bot to the chat but you don’t get any of the benefits of having the chat and group linked. If you then try to relink the chat and the group, you are finally shown this error message making it clear that bots can not be used in a chat that is associated with a Workplace group.
+    
+    <img width="1297" alt="Screen Shot 2022-04-26 at 4 37 32 PM" src="https://user-images.githubusercontent.com/99914387/166559194-41a39093-e283-43ea-8415-7645f4c1351f.png">
+    
+    Fun fact: you can actually get around this by creating a normal group chat → adding the bot to the chat → use the More Options button in the chat to got to Admin Options → click Create Group. Then your bot is in a chat associated with a group and works normally.
+    
+
+## Final Thoughts
+
+ngrok is an incredible tool. I found it confusing at first but once I got the hang of it it was straightforward to use and extremely helpful in testing my code. I will definitely use this tool again next time I am working work webhooks. Workplace custom integrations, on the other hand, I found incredibly frustrating because I know first hand how powerful they can be. Internally at Facebook, they have built out a ton of incredible Workplace tools such as #remindme to remind you of something at a later date, @silent to make your messages not send out a notification, and #share to share posts from one group to another. All of these commands worked in all chats and groups without having to explicitly add a bot. I hope in the future Workplace will ship these features out to other companies and give us the ability to make our own custom integrations just as powerful.
```

Comment:
"next time I am working on webhooks"

These final thoughts make this read more like a medium post than a guide.  Id leave out this commentary here